howtos/diffs/lr-schedule.diff

```diff
@@ -0,0 +1,89 @@
+diff --git a/examples/mnist/train.py b/examples/mnist/train.py
+index 5dc74db..11a5e50 100644
+--- a/examples/mnist/train.py
++++ b/examples/mnist/train.py
+@@ -84,6 +84,16 @@ def create_model(key):
+   model = nn.Model(CNN, initial_params)
+   return model
+ 
++def create_triangular_schedule(lr_min, lr_max, epochs):
++  """return a triangular learning rate schedule"""
++  top = epochs // 2
++  def learning_rate_fn(epoch):
++    if epoch < top:
++      lr = lr_min + epoch/top * (lr_max - lr_min)
++    else:
++      lr = lr_max - (epoch/top - 1)*(lr_max - lr_min)
```

Comment:
1. Please be consistent in your whitespace. Add spaces around `*`.
2. The `- 1` in the arithmetic confuses me (though it's correct). I would write:

`lr = lr_max - ((epoch - top) / top) * (lr_max - lr_min)`